src/OrchardCore/OrchardCore.Setup.Core/SetupService.cs

```diff
@@ -181,7 +181,7 @@ private async Task<string> SetupInternalAsync(SetupContext context)
                 case DbConnectionValidatorResult.InvalidConnection:
                     context.Errors.Add(String.Empty, S["The provided connection string is invalid or server is unreachable."]);
                     break;
-                case DbConnectionValidatorResult.DocumentFound:
+                case DbConnectionValidatorResult.DocumentTableFound:
```

Comment:
You are missing

                case DbConnectionValidatorResult.ShellDescriptorDocumentFound:

As checked in tenant validator

Or remove this check in both if we apply the suggested change in `DbConnectionValidator`, see my related comment.

Reply:
Hummm, that was left out on purpose. It's my understanding that the shell document will exists before the default tenant is created. So when the shell settings document exists, we'll allow the setup to proceed during the setup only. Maybe I am missing the issue. Just so we are on the same page, could you please explain the step that encounter the original issue?

Reply:
As mentioned in the above comment, yes you can let it out and remove it also in tenant validator.

> Or remove this check in both if we apply the suggested change in DbConnectionValidator, see my related comment.

When we use shell settings it explains why the document table may exist and with any table prefix, as some other app level services could add data in the same database and also in the document table if they use yesSql.

So  when the document table exists we don't want to check if the shell settings document exists, but if the ShellDescriptor document exists, meaning that the database with a given prefix is in use. So when the ShellDescriptor (not shell settings) is found the validation should fail, if not found the validation can pass.

Yes the Default tenant has specificities but for most of the things it is a tenant as others. For example the Default tenant can have a non empty prefix and in that case before another tenant was allowed to have an empty prefix, it is only quite recently that this is no longer allowed in some validators.

Reply:
@jtkech I updated the code. So in the SetupService, we don't want to check for `DocumentTableFound` since in this case the table would already exists with the shell settings. However, in the Tenant validator we should check for both `DocumentTableFound` or `ShellDescriptorDocumentFound` to prevent adding new tenants later. I tested it out, but if you see other issue, feel free to directly change the PR.